src/components/setting/languageSelect.js

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import Select from '../toolbox/select';
+import i18n from '../../i18n';
+import languages from '../../../i18n/languages';
+import styles from './setting.css';
+
+export default function LanguageSelect({ t }) {
+  const languageList = Object.keys(languages).map(key => ({
+    value: key,
+    label: t(languages[key].name),
+  }));
+  const activeLanguage = Math.max(0,
+    languageList.map(({ label }) => label).indexOf(i18n.language));
```

Comment:
Since  i18n.language returns the key:
```suggestion
    languageList.map(({ value }) => value).indexOf(i18n.language));
```